ui/public/js/check-for-IE.js

```diff
@@ -0,0 +1,79 @@
+window.onload = function() {
+    const isIE = !!document.documentMode;
```

Comment:
After some searches I think this is the most reliable way of detecting IE, check property [here](https://www.w3schools.com/jsref/prop_doc_documentmode.asp). There is nice analysis done by people from jquery explaining by documentMode is good enough, check it [here](https://github.com/jquery/jquery/issues/4386).